add socket and zod

vaibhav9852 opened the pull request:
This pull request introduces two major changes to the project:

Socket.IO Integration:

Added Socket.IO to the project for real-time communication.
Set up server-side events to listen for connections and handle communication between clients.
Configured client-side Socket.IO to emit events and listen for responses from the server.
This will enable real-time features like live chat, notifications, or live updates, depending on future use cases.
Zod Validation:

Integrated Zod, a TypeScript-first validation library, for input validation.
Added Zod schemas to validate incoming data from requests (e.g., API endpoints).
The validation ensures that the data received from the client is well-formed, reducing the chances of errors and improving data integrity.
Improved error handling for invalid data, providing clear messages to the client.

vaibhav9852 committed:
add scoket and zod

vaibhav9852 committed:
add test case

yashashvi007 commented on backend/src/app.ts:
```diff
@@ -36,52 +39,7 @@ app.use(passport.initialize());
 app.use(passport.session());
  
 
```
remove comments from pr

yashashvi007 commented on backend/src/middleware/zodValidation.middleawre.ts:
```diff
@@ -0,0 +1,52 @@
+      }
+}
+
+
```
why there are spaces

yashashvi007 commented on backend/src/schemas/user.schema.ts:
```diff
@@ -0,0 +1,14 @@
+import {string, z} from 'zod'
+
+export const userSignup = z.object({
```
define error messages also

yashashvi007 commented on backend/__tests__/controller/auth.controller.test.ts:
```diff
@@ -0,0 +1,68 @@
+        expect((await res).body).toHaveProperty('token')
+        expect((await res).body.data).toHaveProperty('id')
+    })
+
```
add tests for wrong credentials

vaibhav9852 committed:
fix socket and update auth and group controller

vaibhav9852 committed:
add socket , zod validation and redux

vaibhav9852 committed:
add authentication feature

vaibhav9852 commented:
This PR introduces comprehensive authentication functionality . The following features have been implemented:

Signup:
Allows users to create an account with email and password.
Sends a verification email upon successful registration.

Login:
Enables registered users to authenticate using their email and password.
Includes validation for incorrect credentials and unverified accounts.
User can login with GitHub

Email Verification:
Implements email-based account verification with a unique token.
Redirects users to the application after successful verification.

Forgot Password:
Provides an interface for users to request a password reset link.
Sends a secure, time-sensitive token to the user's email.

Reset Password:
Allows users to reset their password using the token received in their email.
Includes validation for expired or invalid tokens.

vaibhav9852 committed:
update chat related component

vaibhav9852 commented:
Implemented functionality for users to send and display messages in group chats and individual user chats.

Created the following components:

Message Component:
Fetches messages using TanStack Query (react-query).
Displays the messages in the chat interface.

ChatBox Component:
Handles the display of messages and integrates the list component.
Provides functionality to send messages within a chat box.

List Component:
Displays a list of groups and users available for chatting.
Supports navigation to individual or group chats.

ShowMedia Component:
Displays media files (images, videos, etc.) shared within the chat.

vaibhav9852 committed:
updated backend controller for group exit and delete

vaibhav9852 commented:
Changes Included:
 Added endpoint and logic for exiting a group.
 Implemented group deletion functionality with necessary database updates.
 Added validation and error handling for edge cases (e.g., unauthorized actions).
 Adjusted Prisma schema or service logic if required.

vaibhav9852 committed:
update service with axios instance

vaibhav9852 committed:
updated frotend
